tests/observatory/dags/telescopes/test_geonames.py

```diff
@@ -45,28 +40,46 @@ def __init__(self, *args, **kwargs):
         """
 
         super(TestGeonames, self).__init__(*args, **kwargs)
+        self.project_id = os.getenv('TESTS_GOOGLE_CLOUD_PROJECT_ID')
+        self.data_location = os.getenv('TESTS_DATA_LOCATION')
+        self.all_countries_path = test_fixtures_path('telescopes', 'geonames', 'allCountries.zip')
+        self.fetch_release_date_path = test_fixtures_path('vcr_cassettes', 'geonames', 'fetch_release_date.yaml')
+        self.list_releases_path = test_fixtures_path('vcr_cassettes', 'geonames', 'list_releases.yaml')
 
-        # GeoNames test release
-        self.geonames_test_path = os.path.join(test_fixtures_path(), 'telescopes', 'geonames.txt')
-        self.fetch_release_date_path = os.path.join(test_fixtures_path(), 'vcr_cassettes',
-                                                    'geonames_fetch_release_date.yaml')
+    def test_dag_structure(self):
+        """ Test that the Geonames DAG has the correct structure.
 
-        self.geonames_test_date = pendulum.datetime(year=3000, month=1, day=1)
-        self.geonames_test_download_file_name = 'geonames_3000_01_01.zip'
-        self.geonames_test_decompress_file_name = 'geonames_3000_01_01.txt'
-        self.geonames_test_transform_file_name = 'geonames_3000_01_01.csv.gz'
-        self.geonames_test_decompress_hash = 'de1bf005df4840d16faf598999d72051'
-        self.geonames_test_transform_crc = '26c14e16'
+        :return: None
+        """
 
-        logging.info("Check that test fixtures exist")
-        self.assertTrue(os.path.isfile(self.geonames_test_path))
-        self.assertTrue(self.geonames_test_decompress_hash, _hash_file(self.geonames_test_path, algorithm='md5'))
+        dag = GeonamesTelescope().make_dag()
+        self.assert_dag_structure({
+            'check_dependencies': ['list_releases'],
+            'list_releases': ['download'],
+            'download': ['upload_downloaded'],
+            'upload_downloaded': ['extract'],
+            'extract': ['transform'],
+            'transform': ['upload_transformed'],
+            'upload_transformed': ['bq_load'],
+            'bq_load': ['cleanup'],
+            'cleanup': []
+        }, dag)
```

Comment:
What do you think about using the actual references to these methods? It doesn't look as nice as the strings, but might prevent extra work later if we for example change the method name 'bq_load' in the telescope template.

```
        dag = GeonamesTelescope().make_dag()
        t = GeonamesTelescope
        self.assert_dag_structure({
            t.check_dependencies.__name__: [t.list_releases.__name__],
            t.list_releases.__name__: [t.download.__name__],
            t.download.__name__: [t.upload_downloaded.__name__],
            t.extract.__name__: [t.transform.__name__],
            t.transform.__name__: [t.upload_transformed.__name__],
            t.upload_transformed: [t.bq_load.__name__],
            t.bq_load.__name__: [t.cleanup.__name__],
            t.cleanup.__name__: []
        }, dag)
```

Reply:
I personally like to use strings because it is independent of anything defined in the telescope class, so the test forces you to make sure that the DAG structure is correct if there are changes.